odoo/addons/base/models/ir_rule.py

```diff
@@ -219,24 +226,43 @@ def get_record_description(rec):
             # If the user has access to the company of the record, add this
             # information in the description to help them to change company
             if company_related and 'company_id' in rec and rec.company_id in self.env.user.company_ids:
-                return f'{rec.display_name} (id={rec.id}, company={rec.company_id.display_name})'
-            return f'{rec.display_name} (id={rec.id})'
-
-        records_description = ', '.join(get_record_description(rec) for rec in records_sudo)
-        failing_records = _("Records: %s", records_description)
+                return f'{description}, {rec.display_name} ({model}: {rec.id}, company={rec.company_id.display_name})'
+            return f'{description}, {rec.display_name} ({model}: {rec.id})'
 
-        user_description = f'{self.env.user.name} (id={self.env.user.id})'
-        failing_user = _("User: %s", user_description)
+        failing_records = '\n '.join(f'- {get_record_description(rec)}' for rec in records_sudo)
 
         rules_description = '\n'.join(f'- {rule.name}' for rule in rules)
-        failing_rules = _("This restriction is due to the following rules:\n%s", rules_description)
+        failing_rules = _("Blame the following rules:\n%s", rules_description)
+
         if company_related:
-            failing_rules += "\n\n" + _('Note: this might be a multi-company issue.')
+            msg_heads = {
+                'read':   _("Uh-oh! You've stumbled upon a %s trapped in a multi-company mystery.", description),
+                'write':  _("Uh-oh! You're trying to edit a %s trapped in a multi-company mystery.", description),
+                'create': _("Uh-oh! %s doesn’t have the power to create a record outside of your company’s boundaries.", user_description),
```

Comment:
this is honestly not clear that it is a multi company issue. I would be really confused as a user to see this error